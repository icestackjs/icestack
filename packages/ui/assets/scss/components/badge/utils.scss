@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("badge.selector");
$DefaultMap: inject("badge.defaults.utils");

#{$S} {
  @each $typeName, $map in map.get($DefaultMap, "sizes") {
    &#{getSelector($typeName)} {
      @include resolve(map.get($map, "default"));
    }
  }
}
