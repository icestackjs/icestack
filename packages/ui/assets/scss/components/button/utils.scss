@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("button.selector");
$DefaultMap: inject("button.defaults.utils");

#{$S} {
  @each $size, $map in map.get($DefaultMap, "sizes") {
    &#{getSelector($size)} {
      @include resolve(map.get($map, "default"));
    }
  }

  @each $shape, $map in map.get($DefaultMap, "shapes") {
    &#{getSelector($shape)} {
      @include resolve(map.get($map, "default"));

      @each $size, $m in map.get($map, "sizes") {
        &:where(#{getSelector($size,#{$S}-)}) {
          @include resolve(map.get($m, "default"));
        }
      }
    }
  }
}
