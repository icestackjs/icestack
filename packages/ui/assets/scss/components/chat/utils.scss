@use "../../utils/index.scss" as *;

@include spread(inject("chat.defaults.utils"));
