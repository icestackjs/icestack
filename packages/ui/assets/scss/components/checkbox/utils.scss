@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("checkbox.selector");
$DefaultMap: inject("checkbox.defaults.utils");

@mixin utils {
  @each $s, $v in map.get($DefaultMap, "sizes") {
    &#{getSelector($s)} {
      @include resolve(map.get($v, "default"));
    }
  }
}

#{$S} {
  @include utils;
}
