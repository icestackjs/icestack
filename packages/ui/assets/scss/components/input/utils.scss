@use "../../utils/index.scss" as *;

@include spread(inject("input.defaults.utils"));
