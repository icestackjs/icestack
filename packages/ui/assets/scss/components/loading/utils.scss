@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("loading.selector");
$DefaultMap: inject("loading.defaults.utils");

#{$S} {
  @each $s, $v in map.get($DefaultMap, "sizes") {
    &#{getSelector($s)} {
      @include resolve($v);
    }
  }

  @each $s, $v in map.get($DefaultMap, "shapes") {
    &#{getSelector($s)} {
      @include resolve($v);
    }
  }
}
