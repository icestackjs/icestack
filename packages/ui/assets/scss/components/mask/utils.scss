@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("mask.selector");
$DefaultMap: inject("mask.defaults.utils");

#{$S} {
  @each $s, $v in map.get($DefaultMap, "shapes") {
    &#{getSelector($s)} {
      @include resolve($v);
    }
  }
}
