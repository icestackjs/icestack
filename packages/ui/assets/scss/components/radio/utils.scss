@use "../../utils/index.scss" as *;
@use "sass:map";

$D: inject("radio.defaults.utils");
$S: inject("radio.selector");

#{$S} {
  @each $s, $v in map.get($D, "sizes") {
    &#{getSelector($s)} {
      @include resolve(map.get($v, "default"));
    }
  }
}
