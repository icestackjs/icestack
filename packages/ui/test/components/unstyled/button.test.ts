import { resolve } from './utils'
import { compileScss, resolveTailwindcss } from '@/sass'
describe('button', () => {
  it('snap', async () => {
    const css = await compileScss(resolve('button'))
    expect(css).toMatchSnapshot()
  })

  it('snap resolved', async () => {
    const css = await compileScss(resolve('button'))
    expect(
      await resolveTailwindcss({
        css,
        config: {
          content: [{ raw: '' }],
          theme: {
            extend: {
              borderRadius: {
                btn: 'var(--rounded-box, 1rem)'
              },
              minHeight: {
                12: '3rem'
              }
            }
          }
        }
      })
    ).toMatchSnapshot()
  })
})
