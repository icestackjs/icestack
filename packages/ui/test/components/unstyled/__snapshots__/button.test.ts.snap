// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button > snap 1`] = `
".btn {
  @apply rounded-btn inline-flex flex-shrink-0 cursor-pointer select-none flex-wrap items-center justify-center border-transparent text-center transition duration-200 ease-out;
  @apply min-h-12 h-12 px-4;
  font-size: 0.875rem;
  line-height: 1em;
  /* disabled */
  /* shapes */
}
.btn-disabled, .btn[disabled], .btn:disabled {
  @apply pointer-events-none;
}
.btn-square {
  @apply h-12 w-12 p-0;
}
.btn-circle {
  @apply h-12 w-12 rounded-full p-0;
}

/* radio input and checkbox as button */
.btn:is(input[type=checkbox]),
.btn:is(input[type=radio]) {
  @apply w-auto appearance-none;
}

.btn:is(input[type=checkbox]):after,
.btn:is(input[type=radio]):after {
  @apply content-[attr(aria-label)];
}"
`;

exports[`button > snap resolved 1`] = `
"
.btn {
  display: inline-flex;
  flex-shrink: 0;
  cursor: pointer;
  user-select: none;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  border-radius: var(--rounded-box, 1rem);
  border-color: transparent;
  text-align: center;
  transition-property: color, background-color, border-color, text-decoration-color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
  height: 3rem;
  min-height: 3rem;
  padding-left: 1rem;
  padding-right: 1rem;
  font-size: 0.875rem;
  line-height: 1em
  /* disabled */
  /* shapes */
}
.btn-disabled, .btn[disabled], .btn:disabled {
  pointer-events: none
}
.btn-square {
  height: 3rem;
  width: 3rem;
  padding: 0px
}
.btn-circle {
  height: 3rem;
  width: 3rem;
  border-radius: 9999px;
  padding: 0px
}

/* radio input and checkbox as button */
.btn:is(input[type=checkbox]),
.btn:is(input[type=radio]) {
  width: auto;
  appearance: none
}

.btn:is(input[type=checkbox]):after,
.btn:is(input[type=radio]):after {
  --tw-content: attr(aria-label);
  content: var(--tw-content)
}"
`;
